site/src/pages/TemplatePage/TemplateLayout.tsx

```diff
@@ -68,7 +74,8 @@ export const TemplateLayout: FC<{ children?: JSX.Element }> = ({
     queryKey: ["template", templateName],
     queryFn: () => fetchTemplate(orgId, templateName),
   });
-  const shouldShowInsights = data?.permissions?.canUpdateTemplate;
+  const shouldShowInsights =
+    data?.permissions?.canUpdateTemplate || data?.permissions?.canReadInsights;
```

Comment:
maybe drop a comment here (why 2 permissions instead of 1) as somebody may find it redundant.